From 3b875b2e00701cca88a3acff2ed0b70163147fe4 Mon Sep 17 00:00:00 2001
From: Woo, Insoo <insoo.woo@intel.com>
Date: Wed, 2 Mar 2016 10:23:03 -0800
Subject: [PATCH 094/143] [VPG]: drm/i915 - Allow a render node to use perfmon ioctl

OCL PerfCounter Extension needs perfmon ioctl to send its context
OA reg configuration for per-context perf monitoring.
For this purpose, this patch allows a render node to access the ioctl
interfaces.

Signed-off-by: Woo, Insoo <insoo.woo@intel.com>
---
 drivers/gpu/drm/i915/i915_dma.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_dma.c b/drivers/gpu/drm/i915/i915_dma.c
index 79c3dc1..0979a27 100644
--- a/drivers/gpu/drm/i915/i915_dma.c
+++ b/drivers/gpu/drm/i915/i915_dma.c
@@ -1353,7 +1353,7 @@ const struct drm_ioctl_desc i915_ioctls[] = {
 	DRM_IOCTL_DEF_DRV(I915_GEM_USERPTR, i915_gem_userptr_ioctl, DRM_RENDER_ALLOW),
 	DRM_IOCTL_DEF_DRV(I915_GEM_CONTEXT_GETPARAM, i915_gem_context_getparam_ioctl, DRM_RENDER_ALLOW),
 	DRM_IOCTL_DEF_DRV(I915_GEM_CONTEXT_SETPARAM, i915_gem_context_setparam_ioctl, DRM_RENDER_ALLOW),
-        DRM_IOCTL_DEF_DRV(I915_PERFMON, i915_perfmon_ioctl, DRM_UNLOCKED),
+        DRM_IOCTL_DEF_DRV(I915_PERFMON, i915_perfmon_ioctl, DRM_UNLOCKED|DRM_RENDER_ALLOW),
 };
 
 int i915_max_ioctl = ARRAY_SIZE(i915_ioctls);
-- 
1.7.1

